docs/spec.md

```diff
@@ -6260,6 +6260,26 @@ out-of-bounds accesses, etc. Unless explicitly called out, all these errors
 result in implementation-defined behavior, but this may change in the
 future ([#1157](https://github.com/openxla/stablehlo/issues/1157)).
 
+#### Shape mismatches
+
+StableHLO supports dynamically-shaped tensors. However, shapes have to agree at
+runtime, otherwise the behavior is implementation-defined. StableHLO does not
```

Comment:
Isn't this undefined as opposed to implementation defined? 

For undefined-ness the notion is no defined behavior and the spec offers absolutely no guarantees about what might happen if these are triggered. Here we would like to delegate the ownership of providing those guarantees to the producer of stablehlo.

Moreover, implementation definiteness has some predictability within a particular implementation. This spirit comes from the fact that there are >1 possible behaviors which differ between implementations.  In this case, there is no defined behavior of addition with varying tensors.

Reply:
Thank you for explaining the difference between undefined and implementation defined! When reading some documents about (P2) I saw both terms being used and I thought they meant the same thing: if something is undefined by the spec, then the implementation can do whatever it wants, so it is implementation defined. Turns out, "implementation defined" applies to correct programs and leaves the choice to the implementation, whereas "undefined" applies to incorrect programs. (Implicitly though, what happens may depend on the implementation, which is why I used the two interchangeably at first.)

I changed it to say "undefined", thanks!